airbyte-commons/src/main/java/io/airbyte/commons/json/Jsons.java

```diff
@@ -212,4 +240,45 @@ public DefaultPrettyPrinter withSeparators(final Separators separators) {
 
   }
 
+  public static JsonNode mergeJsons(final ObjectNode mainConfig, final ObjectNode fromConfig) {
+    return mergeJsons(mainConfig, fromConfig, null);
+  }
+
+  public static JsonNode mergeJsons(final ObjectNode mainConfig, final ObjectNode fromConfig, final JsonNode maskedValue) {
+    for (final String key : Jsons.keys(fromConfig)) {
+      if (fromConfig.get(key).getNodeType() == OBJECT) {
+        // nested objects are merged rather than overwrite the contents of the equivalent object in config
+        if (mainConfig.get(key) == null) {
+          mergeJsons(mainConfig.putObject(key), (ObjectNode) fromConfig.get(key), maskedValue);
+        } else if (mainConfig.get(key).getNodeType() == OBJECT) {
+          mergeJsons((ObjectNode) mainConfig.get(key), (ObjectNode) fromConfig.get(key), maskedValue);
+        } else {
+          throw new IllegalStateException("Can't merge an object node into a non-object node!");
+        }
+      } else {
+        if (maskedValue != null && !maskedValue.isNull()) {
+          LOGGER.debug(String.format("Masking instance wide parameter %s in config", key));
+          mainConfig.set(key, maskedValue);
+        } else {
+          if (!mainConfig.has(key) || isSecretMask(mainConfig.get(key).asText())) {
+            LOGGER.debug(String.format("injecting instance wide parameter %s into config", key));
+            mainConfig.set(key, fromConfig.get(key));
+          }
+        }
+      }
+    }
+    return mainConfig;
+  }
+
+  public static JsonNode getSecretMask() {
```

Comment:
knowledge of secrets has nothing to do in a common library. if you actually want to push for it. then it shouldn't be in a function called `mergeJsons`. It will cause it to be miss-use